Add recipe for ob-reticulate

jackkamm opened the pull request:
### Brief summary of what the package does

Integrates the R reticulate package with Org Babel (specifically, ob-R and ob-python). This allows executing python code, and accessing the results, from within an R session.

### Direct link to the package repository

https://github.com/jackkamm/ob-reticulate

### Your association with the package

Maintainer

### Relevant communications with the upstream package maintainer

None needed

### Checklist

Please confirm with `x`:

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses).
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

riscy commented:
checkdoc (using version 0.6.2):
```
Warning (emacs): 
ob-reticulate.el:41: All variables and subroutines might as well have a documentation string
```

Some other notes:
- Possibly your summary line or Commentary should include "Python" somehow to aid in discoverability, but not a blocker
- You shouldn't need to run `declare-function` on `org-babel-python-format-session-value` when you've already imported 'ob-python
- ob-reticulate.el#L37: Avoid adding advice on import; can you suggest this in your `;;; Commentary` or add a minor mode?
- ob-reticulate.el#L22: Prefer `https` over `http` if possible ([why?](https://news.ycombinator.com/item?id=22933774))
- ob-reticulate.el#L55: Can the `format` and `concat` be combined?
- Add a LICENSE file that GitHub can detect (e.g. no markup) if possible
  See: https://github.com/licensee/licensee

jackkamm commented:
Thanks for the code review. I believe I've addressed all issues now. Below are commentary and links to commits.
> checkdoc (using version 0.6.2):

Done: https://github.com/jackkamm/ob-reticulate/commit/56f4d74f5a2218927aeacce61442b373a0a189e3

> Possibly your summary line or Commentary should include "Python" somehow to aid in discoverability, but not a blocker

Done: https://github.com/jackkamm/ob-reticulate/commit/1438fece650e7df1684540103869e1bc9f577522

> You shouldn't need to run declare-function on org-babel-python-format-session-value when you've already imported 'ob-python

Removed: https://github.com/jackkamm/ob-reticulate/commit/0ce305c776424fd34c672d4485ca1b5b283a814d

However, I should note that when I run flycheck with `emacs-lisp` engine, I get a warning:
```
ob-reticulate.el:85:1:Warning: the function
    ‘org-babel-python-format-session-value’ is not known to be defined.
```
No warning occurs when using flycheck with `emacs-lisp-checkdoc` engine, nor when running `package-lint`, `flycheck-package`, `checkdoc`, or `byte-compile-file`. The function is defined in `ob-python` since Org 9.4. Could the flycheck `emacs-lisp` engine be misfiring due to the fact that Emacs is bundled with Org 9.3?

> Avoid adding advice on import; can you suggest this in your ;;; Commentary or add a minor mode?

Added a minor mode: https://github.com/jackkamm/ob-reticulate/commit/9ab16d7b151feaaa80603c89418d0a364af458e5

> Prefer https over http

Done: https://github.com/jackkamm/ob-reticulate/commit/d6dd9d6f82b8d1327de7dd71069efb4ad07952aa

> Can the format and concat be combined?

The concat is needed because of the `when`. The format could be removed, only relying on concat, but I think it's ugly.

However I've tried to improve the clarity by switching the order of concat and format, hope it helps: https://github.com/jackkamm/ob-reticulate/commit/62781917c59ee6d294958a34daba5390db72535d

> Add a LICENSE file that GitHub can detect (e.g. no markup) if possible

Done: https://github.com/jackkamm/ob-reticulate/commit/a5c532b34000ccfee3e827a19fea23fd427b954a

riscy commented:
> Could the flycheck emacs-lisp engine be misfiring due to the fact that Emacs is bundled with Org 9.3?

Hrm, that's unfortunate -- but yeah, I bet that's what's happening.

> I've tried to improve the clarity by switching the order of concat and format.

Thanks, that suggestion is automated on my end and I'm not sure whether I create more grief than it's worth when I suggest it.

In any case, this look good to go.  Thanks!
